lumber-generate.js

```diff
@@ -30,31 +33,45 @@ program
 
   const config = await new CommandGenerateConfigGetter(program).perform();
   let schema = {};
+
   if (program.db) {
-    const connection = await new Database().connect(config);
-    schema = await new DatabaseAnalyzer(connection, config, true).perform();
+    const connectionPromise = new Database().connect(config);
+    spinners.add('database-connection', { text: 'Connecting to database' }, connectionPromise);
+    const connection = await connectionPromise;
+
+    const schemaPromise = new DatabaseAnalyzer(connection, config, true).perform();
+    spinners.add('database-analysis', { text: 'Analyzing your database' }, schemaPromise);
+    schema = await schemaPromise;
   }
 
-  const { envSecret, authSecret } = await new ProjectCreator(logger)
+  const projectCreationPromise = new ProjectCreator(logger)
     .createProject(config.appName, config);
+  spinners.add('project-creation', { text: 'Creating your project on forestadmin' }, projectCreationPromise);
+
+  const { envSecret, authSecret } = await projectCreationPromise;
   config.forestEnvSecret = envSecret;
   config.forestAuthSecret = authSecret;
 
+  const spinner = spinners.add('dumper', { text: 'Creating your project files' });
+  logger.spinner = spinner;
   const dumper = await new Dumper(config);
 
   await P.each(Object.keys(schema), async (table) => {
     await dumper.dump(table, schema[table]);
   });
+  spinner.succeed();
 
   logger.success(`Hooray, ${chalk.green('installation success')}!`);
   await eventSender.notifySuccess();
   process.exit(0);
 })().catch(async (error) => {
-  logger.error(
+  const logs = [
     'Cannot generate your project.',
     'An unexpected error occured. Please create a Github issue with following error:',
-  );
-  logger.log(error);
-  await eventSender.notifyError();
-  process.exit(1);
+    error,
+  ];
+
+  await terminate(1, {
```

Comment:
[Question] I know that the log are not changing anything in term of code execution order. But could we do something to prevent a Forest Admin password prompt in the middle of the execution logs:
<img width="1078" alt="Screenshot 2019-11-05 at 14 27 35" src="https://user-images.githubusercontent.com/664261/68211792-8459b080-ffd8-11e9-8966-e11f0b98c74a.png">

Reply:
Yes we could, I'll create a story for that